library/func_misc.py

```diff
@@ -401,3 +421,51 @@ def Call(self, args):
                 return value.Dict(d)
 
         raise error.TypeErr(val, 'Dict() expected List or Dict', loc.Missing)
+
+
+class Split(vm._Callable):
+
+    def __init__(self, splitter):
+        # type: (split.SplitContext) -> None
+        vm._Callable.__init__(self)
+        self.splitter = splitter
+
+    def Call(self, args):
+        # type: (typed_args.Reader) -> value_t
+        s = args.PosStr()
+
+        ifs = None  # type: str
+        if args.NumPos():
+            ifs = args.PosStr()
+
+        args.Done()
+
+        # mycpp: rewrite list comprehension over sum type
+        l = []  # type: List[value_t]
+        for elem in self.splitter.SplitForWordEval(s, ifs=ifs):
+            l.append(value.Str(elem))
```

Comment:
FYI not all  list comprehensions work in mycpp, but basic ones do

I think this can be

```
strs = self.splitter...
items = [value.Str(s) for s in strs]
return value.List(items)
```

The limitation is that it has to look lke

```
x = [...]
```

not

```
x = value.List([ ... ])  # needs a temp variable
```

Reply:
Ah, yeah I had done this before. The issue was that the type checker was upset about trying to pass the temp var into the `value.List()` constructor because the tempvar was technically `List[value.Str]`. Adding a `type: List[value_t]` annotation to the tempvar fixed that, though!